shinyorg/shiny
# #224 Added iOS 13 check and new token extraction

### Description of Change ###

Implemented update on iOS push notification registration to use the new NSData structure in XCode 11 / iOS 13. Added explicit OS version check to ensure backwards compat.

Pre iOS 13 token example: 
`<965b251c 6cb1926d e3cb366f dfb16ddd e6b9086a 8a3cac9e 5f857679 376eab7C>`

Post iOS 13 token example:
`{length = 32, bytes = 0x965b251c 6cb1926d e3cb366f dfb16ddd ... 5f857679 376eab7c }`

Here are some resources to explain the reason for this change: 
- https://onesignal.com/blog/ios-13-introduces-4-breaking-changes-to-notifications/
- https://forums.developer.apple.com/thread/117545
- https://nshipster.com/apns-device-tokens/#overturned-in-ios-13

### Issues Resolved ### 

- fixes #224 

### API Changes ###

Underlying iOS API change - no public changes.

### Platforms Affected ### 

- iOS

### Behavioral Changes ###

iOS 13 and above apps will now get the correct and new format for push notification tokens.

### Testing Procedure ###

Register for push notifications on a <13 device to ensure the expected old format token is extracted and test >13 device to ensure the new format extracts the token and can register with 3rd party push providers.

### PR Checklist ###
